### DISCUSSION TITLE
set_partial not working

### DISCUSSION THREAD

**Discussion by Miabread** [emilk/egui#1676] [Q&A]
I'm having difficulty displaying a image line by line as data arrives from another thread. Currently trying to use `TextureHandle::set_partial` but only getting a black square. I've verified the image is valid by saving it to a seprate buffer and then to a png and it works fine.
Here's some of the code: 
```rust
eframe::egui::CentralPanel::default().show(ctx, |ui| {
    ui.image(&self.texture, self.texture.size_vec2());

    if let Ok((mut line, current_height)) = self.receiver.try_recv() {
        dbg!(current_height);

        let size = [self.image_info.width, 1];
        let image = ColorImage::from_rgba_unmultiplied(size, &line);

        self.texture.set_partial([0, current_height - 1], image);
    }
});
```

**Comment by emilk** [comment]
You need to allocate the whole texture once up-front (e.g. to `Color32::TRANSPARENT`), and then do partial updates

**Reply by Miabread** [reply]
This is what I use to allocate the texture: 
```rust
let capacity = image_info.height * image_info.width * 4;
let blank = vec![0; capacity];

let size = [image_info.width, image_info.height];
let image = ColorImage::from_rgba_unmultiplied(size, blank.as_slice());
let texture = ctx.egui_ctx.load_texture("render", image);
```
The full file is here: https://github.com/jamesBeeProg/prototypes/blob/main/raytrace/src/main.rs

**Reply by emilk** [reply]
Huh. See if you can make a minimal reproduce and if so, please open an issue!

**Comment by borgmanJeremy** [comment]
I also have this issue on 0.18. I was about to submit a bug for it and then saw the note to test against latest main branch. The issue does not occur on the main branch but I did not bisect the code base to identify the commit that fixed it.

A minimal working example can be found here: https://pastebin.com/j7dGwhQu

Note when testing against the main branch the Texture Filter argument is required.
